dev/integration_tests/android_splash_screens/splash_screen_kitchen_sink/android/app/src/main/java/io/flutter/splash_screen_kitchen_sink/MainActivity.java

```diff
@@ -0,0 +1,190 @@
+package io.flutter.splash_screen_kitchen_sink;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewTreeObserver;
+
+import androidx.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.android.FlutterView;
+import io.flutter.embedding.android.SplashScreen;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.StringCodec;
+import io.flutter.view.FlutterMain;
+
+public class MainActivity extends FlutterActivity {
+  private static final String TAG = "MainActivity";
+
+  private static FlutterEngine flutterEngine;
+
+  static {
+    // Explicitly activates Debug logging for the Flutter Android embedding.
+    io.flutter.Log.setLogLevel(Log.DEBUG);
+  }
+
+  // Sends a JSON-serialized log of test events from Android to Flutter.
+  private BasicMessageChannel<String> testChannel;
+  // Log of splash events that is updated, serialized, and sent to Flutter.
+  private SplashTestLog splashTestLog;
+
+  /**
+   * We explicitly provide a {@code FlutterEngine} so that every rotation does not create a
```

Comment:
eesh, we should make sure to document this in the androidmanifest template (i.e. if you don't let the activity internally handle this, it'll be janky and costly)

Reply:
Sure: https://github.com/flutter/flutter/issues/36659